src/Sema.zig

```diff
@@ -16972,6 +17006,23 @@ fn resolvePeerTypes(
         }
     }
 
+    if (convert_to_const_slice) {
+        // turn *[N]T => []const T
+        const chosen_child_ty = chosen_ty.childType();
+        var info = chosen_ty.ptrInfo();
+        info.data.sentinel = chosen_child_ty.sentinel();
+        info.data.mutable = false;
+        info.data.size = .Slice;
+        info.data.pointee_type = switch (chosen_child_ty.tag()) {
+            .array => chosen_child_ty.elemType2(),
+            .array_u8 => Type.initTag(.u8),
+            .array_u8_sentinel_0 => Type.initTag(.u8),
+            else => unreachable,
+        };
+
+        return Type.ptr(sema.arena, info.data);
```

Comment:
My other idea for this was to add a `toConstSlice` function to `Type` which would basically do this.